autosubmit/history/database_managers/experiment_history_db_manager.py

```diff
@@ -20,433 +20,493 @@
+        pragma_result = self.get_from_statement(
+            self.historicaldb_file_path, statement)
+        if len(pragma_result) <= 0:
+            raise Exception(
+                "Error while getting the pragma version."
+                "This might be a signal of a deeper problem. Review previous errors.")
+        return Models.PragmaVersion(*pragma_result[0]).version
 
 
 class ExperimentHistoryDatabaseManager(Protocol):
-  def initialize(self):...
-  def my_database_exists(self):...
-  def is_header_ready_db_version(self):...
-  def is_current_version(self):...
-  def create_historical_database(self):...
-  def update_historical_database(self):...
-  def get_experiment_run_dc_with_max_id(self):...
-  def register_experiment_run_dc(self, experiment_run_dc):...
-  def update_experiment_run_dc_by_id(self, experiment_run_dc):...
-  def is_there_a_last_experiment_run(self):...
-  def get_job_data_all(self):...
-  def register_submitted_job_data_dc(self, job_data_dc):...
-  def update_job_data_dc_by_id(self, job_data_dc):...
-  def update_list_job_data_dc_by_each_id(self, job_data_dcs):...
-  def get_job_data_dc_unique_latest_by_job_name(self, job_name):...
-  def get_job_data_dcs_last_by_run_id(self, run_id):...
-  def get_job_data_dcs_last_by_wrapper_code(self, wrapper_code):...
-  def get_all_last_job_data_dcs(self):...
-  def update_many_job_data_change_status(self, changes):...
-  def _update_job_data_by_id(self, job_data_dc):...
-  def get_job_data_by_name(self, job_name):...
-  def get_job_data_max_counter(self):...
-  def delete_job_data(self, id_):...
-  def delete_experiment_run(self, run_id):...
+    def initialize(self): ...
+
+    def my_database_exists(self): ...
+
+    def is_header_ready_db_version(self): ...
+
+    def is_current_version(self): ...
+
+    def create_historical_database(self): ...
+
+    def update_historical_database(self): ...
+
+    def get_experiment_run_dc_with_max_id(self): ...
+
+    def register_experiment_run_dc(self, experiment_run_dc): ...
+
+    def update_experiment_run_dc_by_id(self, experiment_run_dc): ...
+
+    def is_there_a_last_experiment_run(self): ...
+
+    def get_job_data_all(self): ...
+
+    def register_submitted_job_data_dc(self, job_data_dc): ...
+
+    def update_job_data_dc_by_id(self, job_data_dc): ...
+
+    def update_list_job_data_dc_by_each_id(self, job_data_dcs): ...
+
+    def get_job_data_dc_unique_latest_by_job_name(self, job_name): ...
+
+    def get_job_data_dcs_last_by_run_id(self, run_id): ...
+
+    def get_job_data_dcs_last_by_wrapper_code(self, wrapper_code): ...
+
+    def get_all_last_job_data_dcs(self): ...
+
+    def update_many_job_data_change_status(self, changes): ...
+
+    def _update_job_data_by_id(self, job_data_dc): ...
+
+    def get_job_data_by_name(self, job_name): ...
+
+    def get_job_data_max_counter(self): ...
+
+    def delete_job_data(self, id_): ...
+
+    def delete_experiment_run(self, run_id): ...
```

Comment:
The function below is still indented with only two spaces. Couldn't we reformat the whole file with 4 spaces?